src/motile_plugin/widgets/tracks_view/tree_widget.py

```diff
@@ -420,14 +468,47 @@ def _set_mode(self, mode: str) -> None:
 
         self.mode = mode
         if mode == "all":
-            self.view_direction = "vertical"
+            if self.feature == "tree":
+                self.view_direction = "vertical"
+            else:
+                self.view_direction = "horizontal"
             df = self.track_df
         elif mode == "lineage":
             self.view_direction = "horizontal"
             self._update_lineage_df()
             df = self.lineage_df
         self.navigation_widget.view_direction = self.view_direction
-        self.tree_widget.update(df, self.view_direction, self.selected_nodes)
+        self.tree_widget.update(
+            df, self.view_direction, self.feature, self.selected_nodes
+        )
+
+    def _set_feature(self, feature: str) -> None:
+        """Set the feature mode to 'tree' or 'area'. For this the view is always horizontal.
```

Comment:
Is there a reason to limit it to being horizontal? Does it add significant code complexity to allow the area in the vertical view as well?

Reply:
Thank you!
In the case of a mix of nodes with and without the area attribute, the NaNs in the track_df are now automatically replaced with zeroes. 
Viewing object size was limited to the horizontal view to avoid confusion between two different plot modes and the horizontal mode is the easiest to look at, but once we have a 'switch orientation' button, we could actually allow both.